README.md

```diff
@@ -20,94 +21,140 @@ You should not use it for anything serious.**
-*  `make test-coverage` (report in `./_coverage/index.html`), or
-*  `make test-coverage.json` (report of per-file summary in `./test-coverage.json`).
+The rest of this guide assumes that you are running commands in the dev container.
 
-For extracting (haddock-style) documentation from the code using dune, type
-*  `make docs` (docs entrypoint: `./_build/default/_doc/_html/index.html`)
+Note that the dev container also includes all language-specific tooling, so you
+can run language-specific build commands as desired (e.g. `dune build .` or
+`poetry run <cmd>`). This can be helpful for ensuring that editor integrations
+function properly (e.g. `ocaml-platform` for VSCode).
+
+## Build
+
+All of the build targets specified in the project [Earthfile](./Earthfile) may
+be built within the dev container (or locally if you have earthly installed).
+Some helpful targets are:
+
+* `earthly +build-ocaml` to build and compile the OCaml code (in
+  [./src](./src)).
+* `earthly +build-ligo` to generate ligo and michelson code (produces local
+  output in `./generated`).
+* `earthly +test` to run the full test suite.
+
+For linting type:
+* `earthly +lint`
+
+For test coverage report using bisect_ppx, type:
+* `earthly +test-coverage` (report in `./_coverage/index.html`)
+
+For extracting (haddock-style) documentation from the code using dune, type:
+* `earthly --artifact +docs/ ocaml-docs/`
+
+For running the end-to-end tests (can take 10s of minutes to run), type:
+* `earthly +test-e2e`
+
+For running mutation tests, type:
+* `earthly +test-mutations`
+  * You may also specify the following build arguments (via `--build-arg`):
+  * `n_mutations` (default=25): Number of mutations to perform
+  * `modules` (default='src/burrow.ml src/checker.ml'): A list of src modules to
+    perform mutations on.
+  * `test_cmd` (default='dune build @run-fast-tests'): The test suite to run.
 
-For running the end-to-end tests, type:
-```console
-$ CHECKER_DIR=$(nix-build -A michelson --arg e2eTestsHack true --no-out-link) python e2e/main.py
-```
 
-## Local Deployment
+## Local Dev Deployment
 
-The contract can be deployed to a local, Docker sandbox run using the provided
-[client library](./client). Note that this workflow has only been tested on
-Linux.
+The contract can be deployed to a local flextesa sandbox using the provided
+[client library](./client).
+
+Ensure that the submodules (ctez in particular) are up-to-date. You will need to
+run this on your host if you don't have GitHub authentication set up in the dev
+container.
 
-First, enter a nix shell:
 ```console
-$ nix-shell
+$ git pull --recurse-submodules
 ```
 
+Then, within the dev container:
+
 Generate the LIGO and Michelson code:
 
 ```console
-$ make build-ligo
+$ earthly +build-ligo
 ```
 
-Ensure that the submodules (ctez in particular) are up-to-date:
+Ensure that the client is up to date:
 ```console
-$ git pull --recurse-submodules
+$ poetry install
```

Comment:
This errors for me. See output here: https://gist.github.com/gkaracha/9a36beae2f4fa5e9e82d5c3b5c11afdb